File: leetcode/twoSum.scala
object Solution {
  def twoSum(nums: Array[Int], target: Int): Array[Int] = {
    nums.zipWithIndex.map{
    case (value, ind2) => (ind2, nums.lastIndexOf(target - value))}
  }.filter(x => x._2 >= 0 && x._2 > x._1).flatMap(x => Array(x._1, x._2))
  def main(args: Array[String]): Unit = {
    twoSum(Array(1, 2, 3, 5), 4);
  }
}

File: leetcode/twoSum.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
	public:
		vector<int> twoSum(vector<int>& nums, int target) {
			for (auto ind = nums.begin(); ind != nums.end(); ind++) {
				auto t = find(ind + 1, nums.end(), target - *ind);	
				if (t != nums.end())
					return vector<int> {static_cast<int>(ind - nums.begin()), static_cast<int>(t - nums.begin())};
			}
			return vector<int> {0, 0};
		}
};

int main() {
	Solution solution;
	vector<int> a{3, 1, 2, 1, 5};
	auto res = solution.twoSum(a, 5);
	cout << res[0] << res[1];
}
